crates/bevy_ui/src/render/mod.rs

```diff
@@ -633,8 +635,23 @@ pub fn extract_text_uinodes(
         if !view_visibility.get() || uinode.size().x == 0. || uinode.size().y == 0. {
             continue;
         }
-        let transform = global_transform.compute_matrix()
-            * Mat4::from_translation(-0.5 * uinode.size().extend(0.));
+
+        let (scale, rotation, translation) = global_transform.to_scale_rotation_translation();
```

Comment:
It should be possible to get a `Affine3A` with `.global_transform.affine()`, then modify the `translation` field, then converting to `Mat4`.

I think it might be less code, and more performant.
```rust
let mut affine_transform = global_transform.affine();
// ...
affine_transform.translation = logical_top_left_nearest_pixel.extend(0.).into();
let transform = Mat4::from(affine_transform);
```

Reply:
Yep, splitting up the `GlobalTransform` is ugly. I forgot about the `affine` method, I'll switch to it. 
I'm not sure there will be a measurable performance difference, since this is only done once per text node, not per glyph.